### PR TITLE
Fix lower/upper-case not to depend on JVM locale

### PR DESCRIPTION
**Pull request by findepi**
Found by https://github.com/apache/iceberg/pull/10518

### CODE REVIEW

**Comment by ajantha-bhat** [comment]
I think this PR also fixes https://github.com/apache/iceberg/issues/9276

**Review by ajantha-bhat — APPROVED** [review]

File: api/src/main/java/org/apache/iceberg/catalog/TableIdentifier.java [hunk 1]
```diff
@@ -80,7 +81,7 @@ public String name() {
   public TableIdentifier toLowerCase() {
     String[] newLevels =
         Arrays.stream(namespace().levels()).map(String::toLowerCase).toArray(String[]::new);
-    String newName = name().toLowerCase();
+    String newName = name().toLowerCase(Locale.ROOT);
```
Comment (nastra) [inline api/src/main/java/org/apache/iceberg/catalog/TableIdentifier.java:84]: I think we should also update `TestTableIdentifier`?

Comment (findepi) [reply api/src/main/java/org/apache/iceberg/catalog/TableIdentifier.java:84]: `TestTableIdentifier` calls `org.apache.iceberg.catalog.TableIdentifier#toLowerCase`
i did not find calls to `String#toLowerCase` in `TestTableIdentifier`

**Comment by findepi** [comment]
there is a conflict (`core/src/test/java/org/apache/iceberg/hadoop/TestStaticTable.java`), rebasing

File: api/src/main/java/org/apache/iceberg/catalog/TableIdentifier.java (hunk 1 above)
Comment (nastra) [reply api/src/main/java/org/apache/iceberg/catalog/TableIdentifier.java:84]: thanks for confirming, I was just quickly checking all codeplaces and didn't realize that it's using `TableIdentifier`

**Review by nastra — APPROVED** [review]

**Review by dimas-b — APPROVED** [review]

**Review by Fokko — APPROVED** [review]

**Comment by Fokko** [comment]
Thanks @findepi for the PR, and thanks @ajantha-bhat, @nastra and @dimas-b for the review 🙌

**Comment by findepi** [comment]
thank you @ajantha-bhat @nastra @dimas-b @Fokko for your reviews!

**Pull request merged**